bot/module/tatsu/wrapper.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+
+import aiohttp
+from ratelimit import limits
+
+import bot.module.tatsu.data_structures as ds
+
+
+class ApiWrapper:
+    def __init__(self, key):
+        self.key = key
+        self.base_url = "https://api.tatsu.gg/v1/"
+        self.headers = {"Authorization": key}
+
+    @limits(calls=60, period=60)
+    async def request(self, url):
+        """Directly interact with the API to get the unfiltered results."""
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url=self.base_url + url, headers=self.headers) as result:
+                if result.status != 200:
+                    return result.raise_for_status()
+                return await result.json()
+
+    @limits(calls=60, period=60)
+    async def patch(self, url, payload):
+        async with aiohttp.ClientSession() as session:
+            async with session.patch(
+                url=self.base_url + url, json=payload, headers=self.headers
+            ) as result:
+                if result.status != 200:
+                    return result.raise_for_status()
+                return await result.json()
+
+    async def get_profile(self, user_id: int) -> ds.UserProfile:
+        """Gets a user's profile. Returns a user object on success."""
+        try:
+            result = await self.request(f"users/{user_id}/profile")
+        except Exception as e:
+            raise e
+        try:
+            subscription_renewal = datetime.datetime.strptime(
+                result.get("subscription_renewal"), "%Y-%m-%dT%H:%M:%SZ"
+            )
+        except Exception:
+            subscription_renewal = None
+        user = ds.UserProfile(
+            avatar_hash=result.get('avatar_hash', None),
+            avatar_url=result.get('avatar_url', None),
+            credits_=result.get('credits', None),
+            discriminator=result.get('discriminator', None),
+            user_id=result.get('id', None),
+            info_box=result.get('info_box', None),
+            reputation=result.get('reputation', None),
+            subscription_type=result.get('subscription_type', None),
+            subscription_renewal=subscription_renewal,
+            title=result.get('title', None),
+            tokens=result.get('tokens', None),
+            username=result.get('username', None),
+            xp=result.get('xp', None),
```

Comment:
Actually `get()` function on dict will return `None` by default if the key does not exist. So writing `result.get('something')` is equal to `result.get('something', None)`

We can refactor them